cli/command/service/scale.go

```diff
@@ -59,6 +70,18 @@ func runScale(dockerCli *command.DockerCli, args []string) error {
 		if err := runServiceScale(dockerCli, serviceID, scale); err != nil {
 			errs = append(errs, fmt.Sprintf("%s: %v", serviceID, err))
 		}
+
+		if options.detach {
+			if !flags.Changed("detach") && versions.GreaterThanOrEqualTo(dockerCli.Client().ClientVersion(), "1.29") {
+				fmt.Fprintln(dockerCli.Err(), "Since --detach=false was not specified, tasks will be scaled in the background.\n"+
```

Comment:
Hm; also (this is gonna be tricky); the warning should no longer be printed if the client talks to docker 17.09 (or whichever version changes the default)